src/sources/kubernetes/file_source_builder.rs

```diff
@@ -0,0 +1,380 @@
+            // exclude kuberenetes and vector logs.
+
+            // This is correct, but on best effort basis filtering out of logs from kuberentes system components.
+            // More specificly, it will work for all Kubernetes 1.14 and higher, and for some bellow that.
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"kube-system_*").into());
+
+            // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+        }
+    }
+}
+
+/// Partial UUID in numerical format (without the '-') as defined by https://tools.ietf.org/html/rfc4122.
+/// Has [0,32] UUID characters.
+#[derive(Clone, Debug, Default)]
+struct PartialUid {
+    partial_uid: String,
+}
+
+impl PartialUid {
+    /// Checks partial_uid for validity, and transforms them to numerical format.
+    fn new(mut partial_uid: String) -> Result<Self, BuildError> {
+        // Check if partial_uid contains only UUID valid characters as defined by https://tools.ietf.org/html/rfc4122.
+        if !partial_uid.chars().all(|c| {
+            c.is_numeric() || ('A'..='F').contains(&c) || ('a'..='f').contains(&c) || c == '-'
+        }) {
+            error!(message = "Configuration 'include_pod_uids' contains an illegal UID.", uid = ?partial_uid);
+            return Err(BuildError::IllegalCharacterInUid {
+                uid: partial_uid.clone(),
+            });
+        }
+
+        // Remove dashes from uid so to always have numerical format
+        // of uid.
+        partial_uid.retain(|c| c != '-');
+
+        if partial_uid.chars().count() > 32 {
+            return Err(BuildError::UidToLarge {
+                uid: partial_uid.clone(),
+            });
+        }
+
+        Ok(Self { partial_uid })
+    }
+
+    /// Transforms partial uid to two globs that match on two UUID forms
+    /// as defined by https://tools.ietf.org/html/rfc4122:
+    ///  - 8char-4char-4char-4char-12char
+    ///  - 32 character number (numerical format)
+    /// If partial_uid has less than 32 UUID characters, if will be filled up
+    /// with "[0-9A-Fa-f]" glob pattern so to match exactly 32 UUID characters.
+    fn uid_globs(&self) -> Vec<String> {
+        let parts = self
+            .partial_uid
+            .chars()
+            .map(|ch| String::from_iter(Some(ch)))
+            .chain((0..32).into_iter().map(|_| "[0-9A-Fa-f]".to_owned()))
+            .collect::<Vec<String>>();
+
+        vec![
+            format!(
+                "{}-{}-{}-{}-{}",
+                parts[0..8].concat(),
+                parts[8..12].concat(),
+                parts[12..16].concat(),
+                parts[16..20].concat(),
+                parts[20..32].concat(),
+            ),
+            parts[0..32].concat(),
+        ]
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
There is a decent amount of logic in this module, I think it would be good to add some more tests around the different possible combination of filters.

Reply:
True.